src/hibernate-resume/hibernate-resume-generator.c

```diff
@@ -239,11 +224,19 @@ static int process_resume(void) {
                 "\n"
                 "[Service]\n"
                 "Type=oneshot\n"
-                "ExecStart=" LIBEXECDIR "/systemd-hibernate-resume %2$s %3$" PRIu64 "\n",
+                "ExecStart=" LIBEXECDIR "/systemd-hibernate-resume '%2$s' '%3$" PRIu64 "'",
                 device_unit,
-                device_escaped,
+                strempty(device_escaped),
```

Comment:
this looks a bit ugly in "ps". maybe use empty_to_dash() here? it's how we usually handle this. And then when parsing you can revrese that with empty_or_dash_to_null()